pydata_center/scripts/run_predictive.py

```diff
@@ -0,0 +1,111 @@
+from data.database import get_all_server_metrics, mark_status
+from django.core.management.base import BaseCommand
+from monitoring.models import PredictionFlag
+from predictive.anomaly import detect_anomaly, is_heartbeat_missing
+from predictive.forecasting import forecast_cpu
+
+from pydata_center.utils.logger import setup_logger
+
+logger = setup_logger()
+
+CPU_THRESHOLD = 90.0
+WINDOW_SIZE = 5
+METRIC_KEYS = [
+    'cpu', 'ram', 'disk', 'load_avg', 'nginx_down_count', 'uptime'
+]
+
+
+def setup_predictive_pipeline():
+    """
+    Step 1: Load all server metric histories.
+    """
+    return get_all_server_metrics()
+
+
+def run_forecasting(server_metrics: dict) -> bool:
+    """
+    Step 2: Forecast CPU load and return True if above threshold.
+    """
+    features = _extract_features(server_metrics)
+    prediction = forecast_cpu(features)
+    if prediction > CPU_THRESHOLD:
+        logger.info(
+            f"Server {server_metrics['id']}: "
+            f'forecasted high CPU load ({prediction:.1f}%)'
+        )
+        return True
+    return False
+
+
+def detect_anomalies_step(server_metrics: dict) -> bool:
+    """
+    Step 3: Run the multivariate anomaly detector.
+    """
+    features = _extract_features(server_metrics)
+    if detect_anomaly(features):
+        logger.info(f"Server {server_metrics['id']}: anomaly detected")
+        return True
+    return False
+
+
+def run_heartbeat_check(server_metrics: dict) -> bool:
+    """
+    Step 4: Check for missing heartbeat.
+    """
+    last_hb = server_metrics.get('last_heartbeat')
+    if is_heartbeat_missing(last_hb):
+        logger.info(f"Server {server_metrics['id']}: missing heartbeat")
+        return True
+    return False
+
+
+def _extract_features(server_metrics: dict) -> list:
+    """
+    Peel off the last WINDOW_SIZE readings of each metric,
+    flatten into a single feature vector.
+    """
+    metric_data = {
+        key: server_metrics.get(key, [])[-WINDOW_SIZE:]
+        for key in METRIC_KEYS
+    }
```

Comment:
If any metric is too short, returns an empty list. Consider logging which metric is missing and why.